### DISCUSSION TITLE
Rotate 2D spatial plot

### DISCUSSION THREAD

**Discussion by millerkaplanlab** [giotto-suite/Giotto#681] [General]
Hello: Just wondering if it is possible to rotate a spatial 2D gene plot.

**Comment by jiajic** [comment]
Hi @millerkaplanlab 
Not just yet through the plotting params, but there are workarounds.
If the gene plot only involves spots colored by gene, you can extract the spatial locations and then use `spin()`

```
library(Giotto)

viz = GiottoData::loadGiottoMini('vizgen')

activeSpatUnit(viz) = 'aggregate'
activeFeatType(viz) = 'rna'

# original
spatFeatPlot2D(viz, feats = featIDs(viz)[1:4])
```
![image](https://github.com/drieslab/Giotto/assets/72078254/818e9f21-c6e3-4275-9590-1be48e2e2b05)

```
# rotated
st = getSpatialLocations(gobject = viz)
st_spin = spin(st, 34) # rotate 34 degrees clockwise
viz = setSpatialLocations(viz, st_spin)
spatFeatPlot2D(viz, feats = featIDs(viz)[1:4])
```
![image](https://github.com/drieslab/Giotto/assets/72078254/2c27e53a-6e72-4f56-89ff-246c6aa7d2a0)

```
# reset
viz = setSpatialLocations(viz, st) # return to original
```
